lib/ai/model/tool.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+)
+
+type Tool interface {
+	Name() string
+	Description() string
+	Run(ctx context.Context, input string) (string, error)
+}
+
+type commandExecutionTool struct{}
+
+type commandExecutionToolInput struct {
+	Command string   `json:"command"`
+	Nodes   []string `json:"nodes"`
+	Labels  []Label  `json:"labels"`
+}
+
+func (c *commandExecutionTool) Name() string {
+	return "Command Execution"
+}
+
+func (c *commandExecutionTool) Description() string {
+	return fmt.Sprintf(`Execute a command on a set of remote hosts based on a set of hostnames or/and a set of labels.
+The input must be a JSON object with the following schema:
+
+%vjson
+{
+	"command": string, \\ The command to execute
+	"nodes": []string, \\ Execute a command on all nodes that have the given hostnames
+	"labels": []{"key": string, "value": string} \\ Execute a command on all nodes that has at least one of the labels
+}
+%v
+`, "```", "```")
+}
+
+func (c *commandExecutionTool) Run(ctx context.Context, input string) (string, error) {
+	return "", trace.NotImplemented("not implemented")
```

Comment:
Why do we need not implemented method here? Can we split the interface or something along those lines?

Reply:
Hey, I added some justification for this decision and some context. WDYT?

Reply:
Works for me.